src/Nethermind/Nethermind.Merge.Plugin/EngineRpcModule.cs

```diff
@@ -53,70 +50,99 @@ public class EngineRpcModule : IEngineRpcModule
             _logger = logManager.GetClassLogger();
         }
 
-        public ResultWrapper<ExecutionStatusResult> engine_executionStatus()
-        {
-            return _executionStatusHandler.Handle();
-        }
+        public ResultWrapper<ExecutionStatusResult> engine_executionStatus() => _executionStatusHandler.Handle();
+
+        public Task<ResultWrapper<ExecutionPayload?>> engine_getPayloadV1(byte[] payloadId) =>
+            _getPayloadHandlerV1.HandleAsync(payloadId);
+
+        public async Task<ResultWrapper<GetPayloadV2Result?>> engine_getPayloadV2(byte[] payloadId) =>
+            await _getPayloadHandlerV2.HandleAsync(payloadId);
 
-        public async Task<ResultWrapper<ExecutionPayloadV1?>> engine_getPayloadV1(byte[] payloadId)
+        public async Task<ResultWrapper<PayloadStatusV1>> engine_newPayloadV1(ExecutionPayload executionPayload)
         {
-            return await (_getPayloadHandlerV1.HandleAsync(payloadId));
+            if (executionPayload.Withdrawals != null)
```

Comment:
would be better if we versioned ExecutionPayload

Reply:
Resolving as a duplicate of https://github.com/NethermindEth/nethermind/pull/4731#discussion_r1061951311